### DISCUSSION TITLE
unable to connect to Zookeeper SSL

### DISCUSSION THREAD

**Discussion by ramamanoj142000** [provectus/kafka-ui#1643] [Q&A] [answered]
Hi 
I'm getting errors while connecting  Kafka-ui on zookeeper-SSL, am i missing something ? . below are detail for docker-compose . Zookeeper host are reachable from this host . 
Docker-compose file 
---
version: '3.4'
services:
  kafka-ui:
    container_name: kafka-ui
    image: docker.apple.com/isds-dataanalytics/isds-provectus-kafka-ui:latest
    ports:
      - 9096:8080
    environment:
       - KAFKA_CLUSTERS_0_NAME= kafkaservice-2
       - KAFKA_CLUSTERS_0_PROPERTIES_SECURITY_PROTOCOL= SASL_SSL
       -  KAFKA_CLUSTERS_0_BOOTSTRAPSERVERS=isds-kafkaservice-dev-west2-lkf01.cognizant.com:9094
       -  KAFKA_CLUSTERS_0_ZOOKEEPER=isds-kafkaservice-dev-west2-lzk01.cognizant.com:2185
       -  KAFKA_CLUSTERS_0_PROPERTIES_SSL_TRUSTSTORE_LOCATION= /kafka.truststore.jks
       -  KAFKA_CLUSTERS_0_PROPERTIES_SSL_TRUSTSTORE_PASSWORD= xxxxxx
       -  KAFKA_CLUSTERS_0_ZOOKEEPER_CLIENTCNXNSOCKETr= org.apache.zookeeper.ClientCnxnSocketNetty
       -  KAFKA_CLUSTERS_0_ZOOKEEPER_CLIENT_SECURE= 'true'
       -  KAFKA_CLUSTERS_0_ZOOKEEPER_SSL_KEYSTORE_LOCATION= /kafka.keystore.jks
       -  KAFKA_CLUSTERS_0_ZOOKEEPER_SSL_KEYSTORE_PASSWORD= xxxxxx
       -  KAFKA_CLUSTERS_0_ZOOKEEPER_SSL_TRUSTSTORE_LOCATION= /kafka.truststore.jks
       -  KAFKA_CLUSTERS_0_ZOOKEEPER_SSL_TRUSTSTORE_PASSWORD= xxxxxx
       -  KAFKA_CLUSTERS_0_PROPERTIES_SASL_MECHANISM=SCRAM-SHA-512
       -  KAFKA_CLUSTERS_0_PROPERTIES_SASL_JAAS_CONFIG=org.apache.kafka.common.security.scram.ScramLoginModule required username="kafka_admin" password="O9csbZJJtGjURn2Akl";

    volumes:
       - /ngs1/keystore/keystore.jks:/kafka.keystore.jks
       - /ngs1/keystore/truststore.jks:/kafka.truststore.jks

===============command to run -- 
    docker-compose  -f docker-compose.yml up

============================

Logs as below-

 kafka-ui    | 2022-02-21 03:35:18,972 INFO  [kafka-admin-client-thread | adminclient-1-SendThread(isds-kafkaservice-dev-west2-lzk01.cognizant.com:2185)] o.a.z.ClientCnxn: SASL config status: Will not attempt to authenticate using SASL (unknown error)
kafka-ui    | 2022-02-21 03:35:18,972 INFO  [kafka-admin-client-thread | adminclient-1-SendThread(isds-kafkaservice-dev-west2-lzk01.cognizant.com:2185)] o.a.z.ClientCnxn: Socket connection established, initiating session, client: /172.18.0.2:41788, server: isds-kafkaservice-dev-west2-lzk01.cognizant.com/100.77.82.57:2185
kafka-ui    | 2022-02-21 03:35:18,973 WARN  [kafka-admin-client-thread | adminclient-1-SendThread(isds-kafkaservice-dev-west2-lzk01.cognizant.com:2185)] o.a.z.ClientCnxn: Session 0x0 for sever isds-kafkaservice-dev-west2-lzk01.cognizant.com/100.77.82.57:2185, Closing socket connection. Attempting reconnect except it is a SessionExpiredException.
kafka-ui    | org.apache.zookeeper.ClientCnxn$EndOfStreamException: Unable to read additional data from server sessionid 0x0, likely server has closed socket
kafka-ui    |   at org.apache.zookeeper.ClientCnxnSocketNIO.doIO(ClientCnxnSocketNIO.java:77)
kafka-ui    |   at org.apache.zookeeper.ClientCnxnSocketNIO.doTransport(ClientCnxnSocketNIO.java:350)
kafka-ui    |   at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1280)
kafka-ui    | 2022-02-21 03:35:19,689 INFO  [kafka-admin-client-thread | adminclient-1-SendThread(isds-kafkaservice-dev-west2-lzk01.cognizant.com:2185)] o.a.z.ClientCnxn: Opening socket connection to serverisds-kafkaservice-dev-west2-lzk01.cognizant.com/100.77.82.57:2185.
kafka-ui    | 2022-02-21 03:35:19,689 INFO  [kafka-admin-client-thread | adminclient-1-SendThread(isds-kafkaservice-dev-west2-lzk01.cognizant.com:2185)] o.a.z.ClientCnxn: SASL config status: Will not attempt to authenticate using SASL (unknown error)
kafka-ui    | 2022-02-21 03:35:19,689 INFO  [kafka-admin-client-thread | adminclient-1-SendThread(isds-kafkaservice-dev-west2-lzk01.cognizant.com:2185)] o.a.z.ClientCnxn: Socket connection established, initiating session, client: /172.18.0.2:41790, server: isds-kafkaservice-dev-west2-lzk01.cognizant.com/100.77.82.57:2185
kafka-ui    | 2022-02-21 03:35:19,690 WARN  [kafka-admin-client-thread | adminclient-1-SendThread(isds-kafkaservice-dev-west2-lzk01.cognizant.com:2185)] o.a.z.ClientCnxn: Session 0x0 for sever isds-kafkaservice-dev-west2-lzk01.cognizant.com/100.77.82.57:2185, Closing socket connection. Attempting reconnect except it is a SessionExpiredException.
kafka-ui    | org.apache.zookeeper.ClientCnxn$EndOfStreamException: Unable to read additional data from server sessionid 0x0, likely server has closed socket
kafka-ui    |   at org.apache.zookeeper.ClientCnxnSocketNIO.doIO(ClientCnxnSocketNIO.java:77)
kafka-ui    |   at org.apache.zookeeper.ClientCnxnSocketNIO.doTransport(ClientCnxnSocketNIO.java:350)
kafka-ui    |   at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1280)
kafka-ui    | 2022-02-21 03:35:20,109 INFO  [kafka-admin-client-thread | adminclient-1-SendThread(isds-kafkaservice-dev-west2-lzk01.cognizant.com:2185)] o.a.z.ClientCnxn: Opening socket connection to serverisds-kafkaservice-dev-west2-lzk01.cognizant.com/100.77.82.57:2185.
kafka-ui    | 2022-02-21 03:35:20,110 INFO  [kafka-admin-client-thread | adminclient-1-SendThread(isds-kafkaservice-dev-west2-lzk01.cognizant.com:2185)] o.a.z.ClientCnxn: SASL config status: Will not attempt to authenticate using SASL (unknown error)
kafka-ui    | 2022-02-21 03:35:20,110 INFO  [kafka-admin-client-thread | adminclient-1-SendThread(isds-kafkaservice-dev-west2-lzk01.cognizant.com:2185)] o.a.z.ClientCnxn: Socket connection established, initiating session, client: /172.18.0.2:41792, server: isds-kafkaservice-dev-west2-lzk01.cognizant.com/100.77.82.57:2185
kafka-ui    | 2022-02-21 03:35:20,111 WARN  [kafka-admin-client-thread | adminclient-1-SendThread(isds-kafkaservice-dev-west2-lzk01.cognizant.com:2185)] o.a.z.ClientCnxn: Session 0x0 for sever isds-kafkaservice-dev-west2-lzk01.cognizant.com/100.77.82.57:2185, Closing socket connection. Attempting reconnect except it is a SessionExpiredException.
kafka-ui    | org.apache.zookeeper.ClientCnxn$EndOfStreamException: Unable to read additional data from server sessionid 0x0, likely server has closed socket
kafka-ui    |   at org.apache.zookeeper.ClientCnxnSocketNIO.doIO(ClientCnxnSocketNIO.java:77)
kafka-ui    |   at org.apache.zookeeper.ClientCnxnSocketNIO.doTransport(ClientCnxnSocketNIO.java:350)
kafka-ui    |   at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1280)
kafka-ui    | 2022-02-21 03:35:20,924 INFO  [kafka-admin-client-thread | adminclient-1-SendThread(isds-kafkaservice-dev-west2-lzk01.cognizant.com:2185)] o.a.z.ClientCnxn: Opening socket connection to serverisds-kafkaservice-dev-west2-lzk01.cognizant.com/100.77.82.57:2185.
kafka-ui    | 2022-02-21 03:35:20,924 INFO  [kafka-admin-client-thread | adminclient-1-SendThread(isds-kafkaservice-dev-west2-lzk01.cognizant.com:2185)] o.a.z.ClientCnxn: SASL config status: Will not attempt to authenticate using SASL (unknown error)
kafka-ui    | 2022-02-21 03:35:20,925 INFO  [kafka-admin-client-thread | adminclient-1-SendThread(isds-kafkaservice-dev-west2-lzk01.cognizant.com:2185)] o.a.z.ClientCnxn: Socket connection established, initiating session, client: /172.18.0.2:41794, server: isds-kafkaservice-dev-west2-lzk01.cognizant.com/100.77.82.57:2185
kafka-ui    | 2022-02-21 03:35:20,926 WARN  [kafka-admin-client-thread | adminclient-1-SendThread(isds-kafkaservice-dev-west2-lzk01.cognizant.com:2185)] o.a.z.ClientCnxn: Session 0x0 for sever isds-kafkaservice-dev-west2-lzk01.cognizant.com/100.77.82.57:2185, Closing socket connection. Attempting reconnect except it is a SessionExpiredException.
kafka-ui    | org.apache.zookeeper.ClientCnxn$EndOfStreamException: Unable to read additional data from server sessionid 0x0, likely server has closed socket
kafka-ui    |   at org.apache.zookeeper.ClientCnxnSocketNIO.doIO(ClientCnxnSocketNIO.java:77)
kafka-ui    |   at org.apache.zookeeper.ClientCnxnSocketNIO.doTransport(ClientCnxnSocketNIO.java:350)
kafka-ui    |   at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1280)
kafka-ui    | 2022-02-21 03:35:21,967 INFO  [kafka-admin-client-thread | adminclient-1-SendThread(isds-kafkaservice-dev-west2-lzk01.cognizant.com:2185)] o.a.z.ClientCnxn: Opening socket connection to serverisds-kafkaservice-dev-west2-lzk01.cognizant.com/100.77.82.57:2185.
kafka-ui    | 2022-02-21 03:35:21,967 INFO  [kafka-admin-client-thread | adminclient-1-SendThread(isds-kafkaservice-dev-west2-lzk01.cognizant.com:2185)] o.a.z.ClientCnxn: SASL config status: Will not attempt to authenticate using SASL (unknown error)
kafka-ui    | 2022-02-21 03:35:21,967 INFO  [kafka-admin-client-thread | adminclient-1-SendThread(isds-kafkaservice-dev-west2-lzk01.cognizant.com:2185)] o.a.z.ClientCnxn: Socket connection established, initiating session, client: /172.18.0.2:41796, server: isds-kafkaservice-dev-west2-lzk01.cognizant.com/100.77.82.57:2185
kafka-ui    | 2022-02-21 03:35:21,968 WARN  [kafka-admin-client-thread | adminclient-1-SendThread(isds-kafkaservice-dev-west2-lzk01.cognizant.com:2185)] o.a.z.ClientCnxn: Session 0x0 for sever isds-kafkaservice-dev-west2-lzk01.cognizant.com/100.77.82.57:2185, Closing socket connection. Attempting reconnect except it is a SessionExpiredException.
kafka-ui    | org.apache.zookeeper.ClientCnxn$EndOfStreamException: Unable to read additional data from server sessionid 0x0, likely server has closed socket
kafka-ui    |   at org.apache.zookeeper.ClientCnxnSocketNIO.doIO(ClientCnxnSocketNIO.java:77)
kafka-ui    |   at org.apache.zookeeper.ClientCnxnSocketNIO.doTransport(ClientCnxnSocketNIO.java:350)
kafka-ui    |   at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1280)
kafka-ui    | 2022-02-21 03:35:22,523 INFO  [kafka-admin-client-thread | adminclient-1-SendThread(isds-kafkaservice-dev-west2-lzk01.cognizant.com:2185)] o.a.z.ClientCnxn: Opening socket connection to serverisds-kafkaservice-dev-west2-lzk01.cognizant.com/100.77.82.57:2185.
kafka-ui    | 2022-02-21 03:35:22,524 INFO  [kafka-admin-client-thread | adminclient-1-SendThread(isds-kafkaservice-dev-west2-lzk01.cognizant.com:2185)] o.a.z.ClientCnxn: SASL config status: Will not attempt to authenticate using SASL (unknown error)
kafka-ui    | 2022-02-21 03:35:22,524 INFO  [kafka-admin-client-thread | adminclient-1-SendThread(isds-kafkaservice-dev-west2-lzk01.cognizant.com:2185)] o.a.z.ClientCnxn: Socket connection established, initiating session, client: /172.18.0.2:41798, server: isds-kafkaservice-dev-west2-lzk01.cognizant.com/100.77.82.57:2185
kafka-ui    | 2022-02-21 03:35:22,525 WARN  [kafka-admin-client-thread | adminclient-1-SendThread(isds-kafkaservice-dev-west2-lzk01.cognizant.com:2185)] o.a.z.ClientCnxn: Session 0x0 for sever isds-kafkaservice-dev-west2-lzk01.cognizant.com/100.77.82.57:2185, Closing socket connection. Attempting reconnect except it is a SessionExpiredException.
kafka-ui    | org.apache.zookeeper.ClientCnxn$EndOfStreamException: Unable to read additional data from server sessionid 0x0, likely server has closed socket
kafka-ui    |   at org.apache.zookeeper.ClientCnxnSocketNIO.doIO(ClientCnxnSocketNIO.java:77)
kafka-ui    |   at org.apache.zookeeper.ClientCnxnSocketNIO.doTransport(ClientCnxnSocketNIO.java:350)
kafka-ui    |   at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1280)
^CGracefully stopping... (press Ctrl+C again to force)
Stopping kafka-ui   ... done

**_I can nc or ping the zookeeper from this host ,_** 

[kafka@isds-kafkaservice-dev-west2-lkf01 03:35:25 /ngs1]$ nc -v 100.77.82.57 2185
Ncat: Version 7.50 ( https://nmap.org/ncat )

**Comment by Haarolean** [comment]
Hi, I noticed a typo in this env. variable:
`KAFKA_CLUSTERS_0_ZOOKEEPER_CLIENTCNXNSOCKETr`
there's an extra 'r' at the end. Let's fix this and try again?

**Reply by ramamanoj142000** [reply]
even after correcting , i'm getting the same error 

-  KAFKA_CLUSTERS_0_ZOOKEEPER_CLIENTCNXNSOCKET= org.apache.zookeeper.ClientCnxnSocketNetty
       -  KAFKA_CLUSTERS_0_ZOOKEEPER_CLIENT_SECURE= 'true'
       -  KAFKA_CLUSTERS_0_ZOOKEEPER_SSL_KEYSTORE_LOCATION= /kafka.keystore.jks
       -  KAFKA_CLUSTERS_0_ZOOKEEPER_SSL_KEYSTORE_PASSWORD= xxx
       -  KAFKA_CLUSTERS_0_ZOOKEEPER_SSL_TRUSTSTORE_LOCATION= /kafka.truststore.jks
       -  KAFKA_CLUSTERS_0_ZOOKEEPER_SSL_TRUSTSTORE_PASSWORD= xxxx
       -  KAFKA_CLUSTERS_0_PROPERTIES_SASL_MECHANISM=SCRAM-SHA-512
  

.ClientCnxn: Session 0x0 for sever isds-kafkaservice-dev-west2-lzk01. cognizant.com/100.77.82.57:2185, Closing socket connection. Attempting reconnect except it is a SessionExpiredException.
kafka-ui    | org.apache.zookeeper.ClientCnxn$EndOfStreamException: Unable to read additional data from server sessionid 0x0, likely server has closed socket
kafka-ui    |   at org.apache.zookeeper.ClientCnxnSocketNIO.doIO(ClientCnxnSocketNIO.java:77)
kafka-ui    |   at org.apache.zookeeper.ClientCnxnSocketNIO.doTransport(ClientCnxnSocketNIO.java:350)
kafka-ui    |   at org.apache.zookeeper.ClientCnxn$SendThread.run(ClientCnxn.java:1280)

**Comment by germanosin** [comment]
Hi, @ramamanoj142000 
Thanks for sharing this issue.

1. Zookeeper connection is not mandatory this product, it will only provide you zookeeper status info.
2. Could you please try with only cluster configs like
```
KAFKA_CLUSTERS_0_NAME=kafkaservice-2
KAFKA_CLUSTERS_0_BOOTSTRAPSERVERS=isds-kafkaservice-dev-west2-lkf01.cognizant.com:9094
KAFKA_CLUSTERS_0_PROPERTIES_SECURITY_PROTOCOL=SASL_SSL
KAFKA_CLUSTERS_0_PROPERTIES_SSL_TRUSTSTORE_LOCATION=/kafka.truststore.jks
KAFKA_CLUSTERS_0_PROPERTIES_SSL_TRUSTSTORE_PASSWORD=xxxx
KAFKA_CLUSTERS_0_PROPERTIES_SSL_KEYSTORE_LOCATION=/kafka.keystore.jks
KAFKA_CLUSTERS_0_PROPERTIES_SSL_KEYSTORE_PASSWORD=xxxx
KAFKA_CLUSTERS_0_PROPERTIES_SSL_KEY_PASSWORD=xxxx
KAFKA_CLUSTERS_0_PROPERTIES_SASL_MECHANISM=SCRAM-SHA-512
KAFKA_CLUSTERS_0_PROPERTIES_SASL_JAAS_CONFIG='org.apache.kafka.common.security.scram.ScramLoginModule required username="kafka_admin" password="O9csbZJJtGjURn2Akl";'
```
3. If this will not work, please provide us logs again

**Reply by ramamanoj142000** [reply]
yes , this is working. but do you know why it is not connecting with the zookeeper? otherwise, i will mark this resolve.

**Reply by germanosin** [reply]
Ok, great. I'll check your zookeeper configs, and will return to you.

**Reply by germanosin** [reply]
@ramamanoj142000 
There is an internal proposal to remove zookeeper configs & status from our product. 
Currently, we use this config only for ZK status checking, and frankly, it is super basic. 
What is your opinion on it? Is it important for you to see ZK connection status?

**Comment by germanosin** [comment]
Btw I see that you are attaching kafka.keystore.jks, but you configure to use SCRAM auth. Do you have both SSL and SCRAM authentications?

**Reply by ramamanoj142000** [reply]
Yes , it is SSL and SCRAM .

**Comment by Haarolean** [comment, marked as answer]
Zookeeper was never required, moreover we've recently dropped support for zookeeper (#1663).
